### PR TITLE
pull request for task 12

### PR DESCRIPTION
**Pull request by prgf87**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -119,6 +117,26 @@ describe("app()", () => {
...
+            expect(article).toHaveProperty("comment_count", "2");
+            expect(article).toHaveProperty("author", expect.any(String));
+            expect(article).toHaveProperty("title", expect.any(String));
+            expect(article).toHaveProperty("body", expect.any(String));
+            expect(article).toHaveProperty("topic", expect.any(String));
+            expect(article).toHaveProperty("created_at", expect.any(String));
+            expect(article).toHaveProperty("votes", expect.any(Number));
+            expect(article).toHaveProperty(
+              "article_img_url",
+              expect.any(String)
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: There are a lot of repeated lines in this test that are the same as the test before it. If this test is only interested in the `comment_count` property then that's the only property you need to test for here 👍

File: controllers/app.controllers.js [hunk 2]
```diff
@@ -30,8 +29,8 @@ const getEndPoints = (_, response, next) => {
 const getArticleById = (request, response, next) => {
   const { article_id } = request.params;
   fetchArticleById(article_id)
-    .then((data) => {
-      response.status(200).send({ article: data });
+    .then((article) => {
+      response.status(200).send({ article: article });
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/app.controllers.js]: this is a nice syntax change! 👌 You go a step further, as it's the same name for both the key and value you can just have 
`.send({ article });`

File: models/app.models.js [hunk 3]
```diff
@@ -16,13 +16,23 @@ const fetchEndPoints = () => {
 
 const fetchArticleById = (id) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = ${id}`)
+    .query(
+      `
+      SELECT articles.*, COUNT(comments.article_id) AS comment_count 
+      FROM articles 
+      JOIN comments ON comments.article_id = articles.article_id
+      WHERE articles.article_id = ${id}
```
Comment (Pipe-Smoking-Rabbit) [inline models/app.models.js]: Big security risk here to the database 🚨 
What if someone puts in a path of _"/api/articles/;DROP+ARTICLES;"_ 
If you're putting values directly into the query they might be read by PSQL and are open to SQL injection, how can we protect against that ?
_hint: $1_

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Very close, there's a suggestion for a change to trim down the code in the test. More importantly though the model is a big security risk to your database currently, have a look at protecting it against injection 🚨

**Comment by Hard-Luck** [comment]
> Very close, there's a suggestion for a change to trim down the code in the test. More importantly though the model is a big security risk to your database currently, have a look at protecting it against injection 🚨

This still doesn't seem to have been addressed

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Still need to fix the security issue with the template string for the id as mitch suggested, look below in the conversation for more info

**Review by Hard-Luck — APPROVED** [review]

**Pull request merged**